From e4e79a17404de599ddf78305c3af09a55979b3e4 Mon Sep 17 00:00:00 2001
From: jsjsjsjsjsjsjson <chara8892@gmail.com>
Date: Sun, 16 Aug 2026 21:17:58 +0800
Subject: [PATCH 1/6] x86/fpu: export fpregs_lock_and_load to GPL modules

The QuarkCompat compatibility layer must operate on the current user x87
stack while handling FCMOV and FCOMI-family #UD exceptions. Export the
existing helper as GPL-only so it can be built as a module.

Signed-off-by: jsjsjsjsjsjsjson <chara8892@gmail.com>
---
 arch/x86/kernel/fpu/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/kernel/fpu/core.c b/arch/x86/kernel/fpu/core.c
index 7526b1688..00b737d01 100644
--- a/arch/x86/kernel/fpu/core.c
+++ b/arch/x86/kernel/fpu/core.c
@@ -834,6 +834,7 @@ void fpregs_lock_and_load(void)
 	if (test_thread_flag(TIF_NEED_FPU_LOAD))
 		fpregs_restore_userregs();
 }
+EXPORT_SYMBOL_GPL(fpregs_lock_and_load);
 
 #ifdef CONFIG_X86_DEBUG_FPU
 /*
